File: HelloAFNetworking/HelloAFNetworking/ViewController.m
//
//  ViewController.m
//  HelloAFNetworking
//
//  Created by Peter Shearer on 9/24/14.
//  Copyright (c) 2014 Peter Shearer. All rights reserved.
//

#import "ViewController.h"
#import "AFNetworking.h"
#import "PlaceholderUser.h"

@interface ViewController ()

@property (nonatomic, strong) NSMutableArray *users;

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
    [self getUsers];
}

- (void) getUsers {
    
    NSString *urlString = @"http://jsonplaceholder.typicode.com/users/";
    NSURL *url = [NSURL URLWithString:urlString];
    NSURLRequest *request = [NSURLRequest requestWithURL:url];
    
    AFHTTPRequestOperation *operation = [[AFHTTPRequestOperation alloc] initWithRequest:request];
    operation.responseSerializer = [AFJSONResponseSerializer serializer];
    
    [operation setCompletionBlockWithSuccess:^(AFHTTPRequestOperation *operation, id responseObject) {
        NSDictionary *dict = (NSDictionary*)responseObject;
        NSMutableArray *array = [[NSMutableArray alloc]init];
        
        for (id innerDict in dict) {
            PlaceholderUser *user = [PlaceholderUser init:innerDict];
            [array addObject:user];
        }
        
        self.users = array;
        [self outputUsers];
        
    } failure:^(AFHTTPRequestOperation *operation, NSError *error) {
        NSLog(@"This failed because %@", error.description);
    }];
    
    [operation start];
    
}

-(void) outputUsers {
    for (id user in self.users) {
        PlaceholderUser *placeholderUser = (PlaceholderUser*)user;
        NSLog(@"%@", placeholderUser);
    }
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

@end

File: SwiftlyNetworking/SwiftlyNetworking/ViewController.swift
//
//  ViewController.swift
//  SwiftlyNetworking
//
//  Created by Peter Shearer on 10/13/14.
//  Copyright (c) 2014 Peter Shearer. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    var users: [PlaceholderUser] = [PlaceholderUser]()
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
        
        var url = NSURL(string: "http://jsonplaceholder.typicode.com/users")
        var request = NSURLRequest(URL: url)
        var operation = AFHTTPRequestOperation(request: request)
        operation.responseSerializer = AFJSONResponseSerializer()
        
        operation.setCompletionBlockWithSuccess({ (operation:AFHTTPRequestOperation!, responseObject:AnyObject!) -> Void in
            let responseDict = responseObject as Array<AnyObject>
            
            for inner in responseDict {
                let innerDict = inner as Dictionary<String, AnyObject>
                var user = PlaceholderUser.initWithDictionary(innerDict)
                self.users.append(user)
            }
            
            self.outputUsers()
            
            }, failure: { (operation: AFHTTPRequestOperation!, error:NSError!) -> Void in
                println(error.description)
        })
        
        operation.start()
    }
    
    func outputUsers() {
        for user in self.users {
            println(user.description())
        }
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }


}

